kratos/spatial_containers/geometrical_objects_bins.h

```diff
@@ -183,30 +183,33 @@ class GeometricalObjectsBins
         }
     }
 
-    /** This method takes a point and finds the nearest object to it.
+    /** This method takes a point and finds the nearest object to it in a given radius.
      * If there are more than one object in the same minimum distance only one is returned
+     * If there are no objects in that radius the 
      * Result contains a flag is the object has been found or not. 
     */
      template<typename TPointType>
-    ResultType SearchNearest(TPointType const& ThePoint) {
+    ResultType SearchNearestInRadius(TPointType const& ThePoint, double Radius) {
 		ResultType current_result;
+        current_result.SetDistance(std::numeric_limits<double>::max());
+
+        double radius_increment = *std::max_element(mCellSizes.begin(), mCellSizes.end());
 
         array_1d< std::size_t, 3 > min_position;
         array_1d< std::size_t, 3 > max_position;
 
-        for(int i = 0; i < 3; i++ ) {
-            min_position[i] = CalculatePosition(ThePoint[i], i);
-            max_position[i] = CalculatePosition(ThePoint[i], i) + 1;
-        }
-
-        current_result.SetDistance(std::numeric_limits<double>::max());
+        for(double current_radius = 0 ; current_radius < Radius + radius_increment ; current_radius += radius_increment){
```

Comment:
Once you find a result on a given "current_radius" and you searched all the cells in that current_radius you can break this loop.

Reply:
The last commit (pushed now) breaks if the result found